src/input/keyboard.c

```diff
@@ -397,19 +397,28 @@ handle_cycle_view_key(struct server *server, struct keyinfo *keyinfo)
 
 	/* cycle to next */
 	if (!keyinfo->is_modifier) {
-		bool back_key = false;
+		int back_key = -1;
 		for (int i = 0; i < keyinfo->translated.nr_syms; i++) {
 			if (keyinfo->translated.syms[i] == XKB_KEY_Up
 					|| keyinfo->translated.syms[i] == XKB_KEY_Left) {
 				back_key = true;
 				break;
 			}
+			if (keyinfo->translated.syms[i] == XKB_KEY_Down
+					|| keyinfo->translated.syms[i] == XKB_KEY_Right) {
+				back_key = false;
+				break;
+			}
+		}
+		bool dir =  server->osd_state.direction;
+		if (back_key >= 0) {
+			/* an arrow key has been hit */
+			dir =  back_key;
+		}
+		if (keyinfo->modifiers & WLR_MODIFIER_SHIFT) {
+			/* Shift held down - change direction */
+			dir = 1 - dir;
```

Comment:
I prefer `dir = !dir`

Reply:
No personal preference, I just want to swap the bool, ie direction

Reply:
if "& !dir" is added after WLR_MODIFIER_SHIFT, then S-A-Tab should work properly, but
then you lose the shift key to reverse direction, when doing PreviousWindow.